src/support_sphere/lib/logic/cubit/signup_cubit.dart

```diff
@@ -47,16 +74,33 @@ class SignupCubit extends Cubit<SignupState> {
   void setValid() => emit(state.copyWith(isValid: true));
   void setInvalid() => emit(state.copyWith(isValid: false));
 
+  /// Sign up with email and password.
   Future<void> signUpWithEmailAndPassword() async {
+    // If the form is invalid, do nothing
     if (!state.isValid) return;
     emit(state.copyWith(status: FormzSubmissionStatus.inProgress));
     try {
       // TODO: Add coupon code check for signup
-      await _authenticationRepository.signUp(
+      final response = await _authenticationRepository.signUp(
         email: state.email,
         password: state.password,
         signupCode: state.signupCode,
       );
+      User? user = response.user;
+      if (user == null) {
+        // If for some reason the user is null, we should set form status to failure
+        emit(state.copyWith(status: FormzSubmissionStatus.failure));
+        return;
+      }
+
+      // Once we have an authenticated user, we can create the user profile and person
+      await _userRepository.createNewUser(
```

Comment:
What, if anything, happens if auth signup succeeds but user creation does not? What should happen? If the same user in that state tries to sign up again, what should happen?

Reply:
That's a great point! I'll punt this to a new issue since I think a graceful handling of this is needed.